mmengine/hooks/test_time_aug_hook.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mmengine.runner import Runner
+
+from mmengine.hooks import Hook
+from mmengine.registry import HOOKS, MODELS, RUNNERS
+
+
+@HOOKS.register_module()
+class PrepareTTAHook(Hook):
+    """Wraps `runner.model` with subclass of :class:`BaseTTAModel` in
+    `before_test`.
+
+    Args:
+        tta_cfg (dict): Config dictionary of the test time augmentation model.
+    """
+
+    def __init__(self, tta_cfg: dict):
+        self.tta_cfg = tta_cfg
+
+    def before_test(self, runner: 'Runner') -> None:
+        """Wraps `runner.model` with the subclass of :class:`BaseTTAModel`.
+
+        Args:
+            runner (Runner): The runner of the testing process.
+        """
+        self.tta_cfg['module'] = runner.model  # type: ignore
+        model = MODELS.build(self.tta_cfg)
+        runner.model = model  # type: ignore
+
+
+# Only used for FSDP TTA.
+def build_runner_with_tta(cfg: dict) -> 'Runner':
```

Comment:
Should this function call `build_runner_from_cfg` to build runner? BTW, should this function be moved to `mmengine/registry/build_functions.py`?

Reply:
`build_runner_with_tta` is a temporary workaround, and its stability cannot be ensured until MMEngine fully supports FSDP. Therefore I temporarily put this function here.

Reply:
OK. We need to add a tip to tell users this is an experimental feature and it may be refactored in the future.